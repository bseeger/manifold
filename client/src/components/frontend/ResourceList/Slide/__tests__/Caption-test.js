jest.mock('velocity-react');

import React from 'react';
import renderer from 'react-test-renderer';
import Caption from '../Caption';
import build from 'test/fixtures/build';

describe("Frontend.ResourceList.Caption component", () => {

  const resource = build.entity.resource("1");

  it('renders correctly', () => {
    const component = renderer.create(
      <Caption
        resource={resource}
        collectionId="1"
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
