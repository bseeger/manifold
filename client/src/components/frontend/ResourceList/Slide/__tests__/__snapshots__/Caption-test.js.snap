exports[`Frontend.ResourceList.Caption component renders correctly 1`] = `
<div
  className="slide-caption">
  <header>
    <h2
      className="resource-title"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Image",
        }
      } />
  </header>
  <div
    className="resource-utility">
    <div
      className="bg-neutral90">
      <button
        className="more-link"
        onClick={[Function]}>
        <span
          className="open-text">
          Read More
        </span>
        <span
          className="close-text">
          Hide Description
        </span>
      </button>
      <a
        className="download-link"
        onClick={[Function]}
        style={Object {}}
        target="_blank">
        Download
        <i
          className="manicon manicon-arrow-down" />
      </a>
      <a
        className="detail-link"
        onClick={[Function]}
        style={Object {}}>
        Details
      </a>
    </div>
  </div>
</div>
`;
